lib/checks/isPolynomialToConstantPower.js

```diff
@@ -0,0 +1,19 @@
+const Node = require('../node');
+
+function isPolynomialToConstantPower(node) {
+  // For the special case when the node is in the form (x + 2)^2
+  // First check if is power node and the exponent is a constant
+  // Then check if the base is a polynomial
+
+  if (!(Node.Type.isOperator(node, '^')
+        && Node.Type.isParenthesis(node.args[0])
+        && Node.Type.isConstant(node.args[1]))) {
+    return false;
+  }
+  const base = node.args[0].content;
+  const isPolynomial = base.args.some(Node.PolynomialTerm.isPolynomialTerm);
```

Comment:
Will this let too many things through? This would include things like: 5*sqrt(4)*x and y/(5+x) and all sorts of things that aren't a sum/difference of polynomial and/or constant terms. Maybe that's fine, I'm not sure? To be safer, we could check that all args are either polynomials or constants, and that the root operator is +.